packages/host/tests/integration/search-index-test.gts

```diff
@@ -3122,6 +3216,7 @@ posts/ignore-me.json
                 title: 'post 1',
                 author: {
                   firstName: 'A',
+                  lastName: null,
```

Comment:
We have some fundamental challenges searching for missing fields with SQLite. the way the `json_tree()` function works is that each item in the JSON object is a row in the result set (we can ellide rows that correspond to paths that we don't care about to make things more performant). the desired search result will be one of these rows (via a cross join)--we use the field's JSON path in the search doc to understand the context of the value that we are matching against. The issue is, if a field is simply missing, then there is no row in the result set to test our predicate against. The best we can do is search for an empty field, `null`--those we can find. But a missing field (bascially an `undefined`) is invisible to `json_tree()`. I think we might need a new Query filter `missing` to handle this. there are probably things that we can do in SQLite to find these items, but the structure of the query is gonna be very different

Reply:
Conversely, we might decide that missing and empty are essential equivalent, and in that case we could assert any missing fields are actually `null` in our search docs. That actually is probably more in line with how the tests are written right now.

Reply:
I created ticket CS-6762 to track this work

Reply:
I think treating missing and empty as equivalent seems OK